src/algebra/order/floor.lean

```diff
@@ -214,6 +214,68 @@ ext $ λ x, ceil_eq_iff hn
 
 /-! #### Intervals -/
 
+lemma floor_mem_Icc_of_Icc {a b : ℕ} {x : α} (hx : x ∈ set.Icc (a : α) ↑b) :
```

Comment:
```suggestion
lemma floor_mem_Icc_of_Icc {a b : ℕ} {x : α} (hx : x ∈ set.Icc (a : α) (b : α)) :
```
For coherence, could you use the same type ascription for `a` and `b` (here and in all lemmas below?). At first, I was led to wonder if they were playing different roles, so anything reducing confusion can only help.

Also, the name `floor_mem_Icc_of_mem_Icc` would probably be better.

Reply:
Sure on the type ascription.

And of course on the name. Just a note that the name of the lemma was originally the one you suggested, but I changed it to be in line with the lemma `mem_Ixx_of_Iyy` which is the central lemma in the proofs, figuring that was what I should be mirroring.